### ISSUE TITLE
[Bug]: Fetch failing for Blacktown City council NSW Australia

### ISSUE THREAD

**Issue by mannigill1** [mampfes/hacs_waste_collection_schedule#1708]
### I Have A Problem With:

A specific source

### What's Your Problem

I have configured integration for Blacktown city council using the same config as shown example on github but its failing with data fetch error. 

### Source (if relevant)

1.45.0

### Logs

```Shell
2024-01-22 12:45:47.467 ERROR (SyncWorker_5) [waste_collection_schedule.source_shell] fetch failed for source Blacktown City Council (NSW):
Traceback (most recent call last):
  File "/config/custom_components/waste_collection_schedule/waste_collection_schedule/source_shell.py", line 134, in fetch
    entries = self._source.fetch()
              ^^^^^^^^^^^^^^^^^^^^
  File "/config/custom_components/waste_collection_schedule/waste_collection_schedule/source/blacktown_nsw_gov_au.py", line 73, in fetch
    data = json.loads(r.text)
           ^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/json/__init__.py", line 346, in loads
    return _default_decoder.decode(s)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/json/decoder.py", line 337, in decode
    obj, end = self.raw_decode(s, idx=_w(s, 0).end())
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/json/decoder.py", line 355, in raw_decode
    raise JSONDecodeError("Expecting value", s, err.value) from None
json.decoder.JSONDecodeError: Expecting value: line 1 column 1 (char 0)
```

### Relevant Configuration

```YAML
waste_collection_schedule:
  sources:
    - name: blacktown_nsw_gov_au
      args:
        post_code: 2770
        suburb: Emerton
        street_name: Helena Ave
        street_number: 80
```

### Checklist Source Error

- [X] Use the example parameters for your source (often available in the documentation) (don't forget to restart Home Assistant after changing the configuration)
- [X] Checked that the website of your service provider is still working
- [X] Tested my attributes on the service provider website (if possible)
- [X] I have tested with the latest version of the integration (master) (for HACS in the 3 dot menu of the integration click on "Redownload" and choose master as version)

### Checklist Sensor Error

- [X] Checked in the Home Assistant Calendar tab if the event names match the types names (if types argument is used)

### Required

- [X] I have searched past (closed AND opened) issues to see if this bug has already been reported, and it hasn't been.
- [X] I understand that people give their precious time for free, and thus I've done my very best to make this problem as easy as possible to investigate.

**Comment by 5ila5** [comment]
testing this I sometimes got an incapsular page (tool that prevents scraping). it then worked again. If they "protect" this by using incapsular there is not that much we can do to prevent this. We Could modify the HEADER to be a more "realistic" web browser.

**Comment by mannigill1** [comment]
> testing this I sometimes got an incapsular page (tool that prevents scraping). it then worked again. If they "protect" this by using incapsular there is not that much we can do to prevent this. We Could modify the HEADER to be a more "realistic" web browser.

Thanks for looking into this.

**Comment by 5ila5** [comment]
Probably fixed in #1712

This will be part of release 1.46.0 or is available now as master (git) version (3-dot menu -> redownload -> change version to master -> fully restart HA (this will increase update cycle by a lot)
